Gems/Atom/RHI/Metal/Code/Source/RHI/Device.cpp

```diff
@@ -77,7 +78,16 @@ namespace AZ
             
             m_samplerCache = [[NSCache alloc]init];
             [m_samplerCache setName:@"SamplerCache"];
-            
+
+            m_mainDisplayRefreshRate = Platform::GetMainDisplayRefreshRateInternal();
+
+            // Assume 60hz if 0 is returned.
+            // This can happen on OSX. In future we can hopefully use maximumFramesPerSecond which wont have this issue
+            if (m_mainDisplayRefreshRate < 0.1f)
```

Comment:
Why not just check equality with `0.0f`?

Reply:
In general, I try to avoid == checks for floats. Sometimes == 0.0f works if you are just checking if a value was changes after initializing it to 0.0f but in this case I may get a weird value like -0.0f from the driver and it would fail the 'if' test. Hence it feels more safer to use < 0.1.

Reply:
Equality checks are important in cases where we actually expect a particular value to have logical significance. Here the MacOS specifies that it will return exactly `0.0` as a double (which casts perfectly to `0.0f`) in the case you've identified. I personally don't like magical fudge-factors especially if they aren't correlated with some actual numerical significance.

Reply:
This was addressed by switching the type to int32_t to match other backends.